src/HID-APIs/DefaultKeyboardAPI.hpp

```diff
@@ -42,28 +42,35 @@ size_t DefaultKeyboardAPI::set(KeyboardKeycode k, bool s)
 	}
 	// Its a normal key
 	else{
+        const int numKeycodeSlots = sizeof(_keyReport.keycodes) / sizeof(_keyReport.keycodes[0]);
 		// Add k to the key report only if it's not already present
 		// and if there is an empty slot. Remove the first available key.
-		for (uint8_t i = 0; i < sizeof(_keyReport.keycodes); i++)
-		{
-			auto key = _keyReport.keycodes[i];
-			
-			// Is key already in the list or did we found an empty slot?
-			if (s && (key == uint8_t(k) || key == KEY_RESERVED)) {
-				_keyReport.keycodes[i] = k;
-				return 1;
-			}
-			
-			// Test the key report to see if k is present. Clear it if it exists.
-			if (!s && (key == k)) {
-				_keyReport.keycodes[i] = KEY_RESERVED;
-				return 1;
-			}
-		}
+        uint8_t emptySlot = numKeycodeSlots;
+        uint8_t foundSlot = numKeycodeSlots;
+        for (uint8_t i = 0; i < numKeycodeSlots; i++)
```

Comment:
In this case the new key will be added to the end of the report, not the beginning. If counting downwards has no disadvantage (in code size), it could make more sense to do it this way. This is how other implementations work, I just try to avoid other problems, especially with older/bios systems. But since you tested this, it should be okay, also from the spec. Just as a personal thought...